geowebcache/arcgiscache/src/main/java/org/geowebcache/arcgis/layer/ArcGISCacheLayer.java

```diff
@@ -244,7 +244,7 @@ protected boolean initializeInternal(GridSetBroker gridSetBroker) {
         final GridSetBuilder gsBuilder = new GridSetBuilder();
         GridSet gridSet = gsBuilder.buildGridset(layerName, info, layerBounds);
 
-        getGridsetConfiguration(gridSetBroker).addInternal(gridSet);
+        gridSetBroker.put(gridSet);
```

Comment:
I am not that familiar with how grid sets are managed, is this adding the grid set publicly? rather than internally ...

Reply:
This change simply reverts the line back to its prior version, reverting the breaking change to this line [here](https://github.com/GeoWebCache/geowebcache/commit/7eb3d371b0ba64dafe31ce9361385e326c36fb6a).

### Fuller background:

Testing with different GWC versions, we'd seen that the GWC ArcGIS cache steps in the official docs ([here](https://www.geowebcache.org/docs/latest/configuration/layers/arcgistilingschemes.html)) had stopped working after GeoWebCache 1.10.5 and continued to not work up through the current version, which we've also tested with.

A related [GWC issue](https://github.com/GeoWebCache/geowebcache/issues/696) was opened by others in 2018 reflecting the same GWC ArcGIS cache layer error we've seeing in testing versions up to this point and which our merge request hopefully can be used to finally close out.

Since ArcGIS Cache Layers were working in GWC 1.10.5, we'd used that to track down the code change that caused the issue ([here](https://github.com/GeoWebCache/geowebcache/commit/7eb3d371b0ba64dafe31ce9361385e326c36fb6a)) and reverted the breaking change with our merge request.

The intention of the original breaking change itself is unfortunately unclear with its general branch description "Working on Gridsets and ArcGIS Cache for configuration changes", sparse comments and the direct outcome being ArcGIS Cache Layers themselves no longer working. As its evident the updates were made without testing ArcGIS Cache Layers, we're hoping our new test file can help prevent this going forward.

In short, this reversion allows the grid set configuration to be found for the ArcGIS Cache Layer like in prior GWC versions instead of throwing the "No ArcGISCacheGridsetConfiguration" can be found exception.

Reply:
Thanks for the background :)